components/Authentication/signup.jsx

```diff
@@ -60,107 +53,149 @@ const ShowText = styled.Text`
   padding-right: 15px;
 `;
 
+const ErrorText = styled.Text`
+  font-family: 'Montserrat_600SemiBold';
+  color: red;
+  font-size: 12px;
+`;
+
+const ViewFiller = styled.View`
+  height: 15px;
+`;
+
 const Signup = ({ navigation }) => {
   const [name, setName] = React.useState('');
+  const [validName, setValidName] = React.useState(false);
   const [email, setEmail] = React.useState('');
+  const [validEmail, setValidEmail] = React.useState(false);
   const [password, setPassword] = React.useState('');
+  const [validPassword, setValidPassword] = React.useState(false);
   const [hidePassword, setHidePassword] = React.useState(true);
+  const isFirstRunName = React.useRef(true);
+  const isFirstRunEmail = React.useRef(true);
+  const isFirstRunPassword = React.useRef(true);
+
+  function ValidateEmail(mail) {
+    // eslint-disable-next-line
+    const re = /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+    if (re.test(mail)) {
+      return (true);
+    }
+    return (false);
+  }
+
+  React.useEffect(() => {
+    if (isFirstRunName.current) {
+      isFirstRunName.current = false;
+      return;
+    }
+    setValidName(!((name.length > 1)));
```

Comment:
if you put in else around each of these "actions", then you could combine the three useEffectes into one while still having the same functionality.

Reply:
In this case tho, I think it's better to have multiple useEffects like you have. Ignore the comment above.

Reply:
yeah cuz they're only triggering when the specific field they correlate with is changed, whereas if they were all in one they'd all trigger from a single of the textinputs